wasm2c/wasm-rt.h

```diff
@@ -81,6 +81,10 @@ extern "C" {
 
 #ifdef _MSC_VER
 #define WASM_RT_THREAD_LOCAL __declspec(thread)
+// We use __thread on POSIXy systems. This is disabled on Apple systems right
+// now due to sporadic test failures.
+#elif (defined(__GNUC__) || defined(__clang__)) && !defined(__APPLE__)
+#define WASM_RT_THREAD_LOCAL __thread
 #elif defined(WASM_RT_C11_AVAILABLE)
 #define WASM_RT_THREAD_LOCAL _Thread_local
```

Comment:
Should we prefer the standards based `_Thread_local` when it is available?  (i.e. put it first in this chain?)

Reply:
Yup, makes sense. Will make this change.